pkg/operator/collection.go

```diff
@@ -138,9 +141,16 @@ func (r *collectionReconciler) Reconcile(ctx context.Context, req reconcile.Requ
 	if err := r.ensureCollectorConfig(ctx, &config.Collection); err != nil {
 		return reconcile.Result{}, errors.Wrap(err, "ensure collector config")
 	}
-	if err := r.updateCRDStatus(ctx); err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "update crd status")
+
+	// Reconcile any status updates.
+	for _, obj := range r.statusUpdates {
+		if err := r.client.Status().Update(ctx, obj); err != nil {
+			logger.Error(err, "update status")
```

Comment:
I think referencing the object it failed for would be good as I'm not sure that the error will contain that info by default.